realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,210 @@
+        // FIXME Why? We cannot update a sync schema "destructively"? -> Update Doc
+        //  Shouldn't this be UnsupportedOperation to be consistent with ex. addField, etc.
+        assertFailsWith<java.lang.IllegalArgumentException> {
+            realm.schema.remove(className)
+        }
+        realm.cancelTransaction()
+        Assert.assertTrue(realm.schema.contains(className))
+        realm.close()
+    }
+
+    @Test
+    fun allow_createClass() {
+        val realm = DynamicRealm.getInstance(config!!)
+        val className = "Dogplace"
+        realm.beginTransaction()
+        realm.schema.create("Dogplace")
+        realm.commitTransaction()
+        Assert.assertTrue(realm.schema.contains(className))
+        realm.close()
+    }
+
+    @Test
+    fun disallow_renameClass() {
+        // Init schema
+        Realm.getInstance(config!!).close()
+        DynamicRealm.getInstance(config!!).use { realm ->
+            val existingClass = "StringOnly"
+            val newClass = "Dogplace"
+            realm.beginTransaction()
+            assertFailsWith<java.lang.IllegalArgumentException> {
+                // FIXME Why? We cannot update a sync schema "destructively"? -> Update Doc
+                //  Shouldn't this be UnsupportedOperation to be consistent with ex. addField, etc.
+                realm.schema.rename(existingClass, newClass)
+            }
+            Assert.assertFalse(realm.schema.contains(newClass))
+            realm.cancelTransaction()
+            Assert.assertTrue(realm.schema.contains(existingClass))
+            Assert.assertFalse(realm.schema.contains(newClass))
+        }
+    }
+
+    @Test
+    fun disallow_removeField() {
+        // Init schema
+        Realm.getInstance(config!!).close()
+        val realm = DynamicRealm.getInstance(config!!)
+        val className = "StringOnly"
+        val fieldName = "chars"
+        val objectSchema = realm.schema[className]
+        Assert.assertNotNull(objectSchema)
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.beginTransaction()
+        assertFailsWith<IllegalArgumentException> {
+            // FIXME Why? We cannot update a sync schema "destructively"? -> Update Doc
+            //  Shouldn't this be UnsupportedOperation to be consistent with ex. addPrimaryKey, etc.
+            objectSchema.removeField(fieldName)
+        }
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.cancelTransaction()
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.close()
+    }
+
+    @Test
+    fun allow_addField() {
+        // Init schema
+        Realm.getInstance(config!!).close()
+        val className = "StringOnly"
+        val realm = DynamicRealm.getInstance(config!!)
+        val objectSchema = realm.schema[className]
+        Assert.assertNotNull(objectSchema)
+        realm.beginTransaction()
+        objectSchema!!.addField("foo", String::class.java)
```

Comment:
In other tests, we have done `val objectSchema = realm.schema[className]!!`. This triggers the assertion closer to where the error is instead of the first time the schema is being used. So it makes the error more clear IMO

You also avoid having to use `!!` in all the places the objectSchema is accessed. E.g. the test below checks 3 times for this.